RUFAS/biophysical/animal/reproduction/reproduction.py

```diff
@@ -1,2 +1,2768 @@
+                self._calculate_conception_rate_on_ai_day()
+                self.repro_state_manager.enter(ReproStateEnum.AFTER_AI)
+                reproduction_data_stream.events.add_event(
+                    reproduction_data_stream.days_born,
+                    time.simulation_day,
+                    f"Current repro state(s): {self.repro_state_manager}",
+                )
+                reproduction_data_stream = self._perform_ai(reproduction_data_stream, time.simulation_day)
+
+            reproduction_data_stream = self.cow_pregnancy_update(reproduction_data_stream, time.simulation_day)
+
+        reproduction_data_stream = self._check_do_not_breed_flag(time.simulation_day, reproduction_data_stream)
+
+        return reproduction_data_stream
+
+    def cow_give_birth(self, reproduction_data_stream: ReproductionDataStream, time: Time) -> ReproductionDataStream:
+        """
+        Handle the birth of a calf, resetting reproduction states and updating outputs.
+
+        Parameters
+        ----------
+        reproduction_data_stream : ReproductionDataStream
+            Current reproduction outputs.
+        time : Time
+            The current time in the simulation.
+
+        Returns
+        -------
+        ReproductionDataStream
+            Updated reproduction outputs after birth.
+        """
+        self.repro_state_manager.reset()
+        self.calves += 1
+        reproduction_data_stream.days_in_milk = 1
+        reproduction_data_stream.days_in_pregnancy = 0
+        self.gestation_length = 0
+
+        self.body_weight_at_calving = reproduction_data_stream.body_weight
+
+        reproduction_data_stream.events.add_event(reproduction_data_stream.days_born, time.simulation_day,
+                                              animal_constants.NEW_BIRTH)
+        reproduction_data_stream.events.add_event(
+            reproduction_data_stream.days_born, time.simulation_day,
+            f"{animal_constants.NUM_CALVES_BORN_NOTE}: {self.calves}")
+
+        if self.cow_reproduction_program != AnimalConfig.cow_reproduction_program:
+            reproduction_data_stream = self._set_cow_reproduction_program(
+                reproduction_data_stream,
+                time.simulation_day,
+                CowReproductionProtocol(AnimalConfig.cow_reproduction_program)
+            )
+            self.repro_state_manager.reset()
+
+        if self.cow_reproduction_program in [
+            CowReproductionProtocol.ED,
+            CowReproductionProtocol.ED_TAI,
+        ]:
+            reproduction_data_stream = self._simulate_estrus(
+                reproduction_data_stream,
+                reproduction_data_stream.days_born,
+                time.simulation_day,
+                f"{animal_constants.ESTRUS_AFTER_CALVING_NOTE}: {animal_constants.ESTRUS_DAY_SCHEDULED_NOTE}",
+                AnimalConfig.average_estrus_cycle_return,
+                AnimalConfig.std_estrus_cycle_return,
+            )
```

Comment:
Move to helper function to help manage size of this function?
```suggestion
        def _simulate_estrus_if_eligible(self, reproduction_data_stream: ReproductionDataStream, time: Time) -> ReproductionDataStream:
    if self.cow_reproduction_program in {CowReproductionProtocol.ED, CowReproductionProtocol.ED_TAI}:
        return self._simulate_estrus(
            reproduction_data_stream,
            reproduction_data_stream.days_born,
            time.simulation_day,
            f"{animal_constants.ESTRUS_AFTER_CALVING_NOTE}: {animal_constants.ESTRUS_DAY_SCHEDULED_NOTE}",
            AnimalConfig.average_estrus_cycle_return,
            AnimalConfig.std_estrus_cycle_return,
        )
    return reproduction_data_stream

```